tests/src/core/resolve.js

```diff
@@ -21,34 +21,48 @@ describe('resolve', function () {
     const testContext = utils.testContext({ 'import/resolver': './foo-bar-resolver-v1' });
 
     expect(resolve( '../files/foo'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('foo.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('foo.js'); } }),
     )).to.equal(utils.testFilePath('./bar.jsx'));
 
     expect(resolve( '../files/exception'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('exception.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('exception.js'); } }),
     )).to.equal(undefined);
 
     expect(resolve( '../files/not-found'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('not-found.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('not-found.js'); } }),
     )).to.equal(undefined);
   });
 
   it('resolves via a custom resolver with interface version 1 assumed if not specified', function () {
     const testContext = utils.testContext({ 'import/resolver': './foo-bar-resolver-no-version' });
 
     expect(resolve( '../files/foo'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('foo.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('foo.js'); } }),
     )).to.equal(utils.testFilePath('./bar.jsx'));
 
     expect(resolve( '../files/exception'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('exception.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('exception.js'); } }),
     )).to.equal(undefined);
 
     expect(resolve( '../files/not-found'
-      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('not-found.js'); } }),
+      , Object.assign({}, testContext, { getPhysicalFilename: function () { return utils.getFilename('not-found.js'); } }),
     )).to.equal(undefined);
   });
 
+  it('falls back to getFilename() when getPhysicalFileName() is not available (ESLint < 7.28)', function () {
+    const testContext = utils.testContext({ 'import/resolver': './foo-bar-resolver-v1' });
+
+    expect(resolve( '../files/foo'
+      , Object.assign({}, testContext, { getFilename: function () { return utils.getFilename('foo.js'); } }),
+    )).to.equal(utils.testFilePath('./bar.jsx'));
+
+    expect(resolve( '../files/foo'
+      , Object.assign({}, testContext, { getFilename: function () { throw new Error('Should call getPhysicalFilename() instead of getFilename()'); }, getPhysicalFilename: function () { return utils.getFilename('foo.js'); } }),
+    )).to.equal(utils.testFilePath('./bar.jsx'));
+
+
+  });
```

Comment:
Added this test to make sure the fallback behavior is working. Otherwise the tests have been updated to use `getPhysicalFilename()` instead of `getFilename()`.